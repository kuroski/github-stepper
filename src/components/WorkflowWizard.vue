<template>
  <div data-testid="workflow-wizard">
    <div v-for="step in steps" :key="step.link">
      <span v-if="step.disabled">{{ step.title }}</span>
      <LocalizedLink
        v-else
        :to="{ name: step.link }"
        active-class="workflow-wizzard__step--active"
      >
        {{ step.title }}
      </LocalizedLink>
    </div>
  </div>
</template>

<script>
import { array, guard, object, string, boolean, nullable } from "decoders";
import LocalizedLink from "@/components/LocalizedLink";

const stepDecoder = object({
  title: string,
  link: string,
  disabled: boolean
});

const stepsDecoder = guard(nullable(array(stepDecoder)));

export default {
  name: "WorkflowWizard",
  components: {
    LocalizedLink
  },
  props: {
    steps: {
      type: Array,
      required: true,
      validator: stepsDecoder
    }
  }
};
</script>
