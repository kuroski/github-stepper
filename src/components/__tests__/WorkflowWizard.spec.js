import { renderWithDependencies } from "@unit/helpers";
import WorkflowWizard from "@/components/WorkflowWizard";

describe("WorkflowWizard", () => {
  it("renders a empty list of steps", () => {
    const { html, getByTestId } = renderWithDependencies(WorkflowWizard, {
      props: {
        steps: []
      }
    });

    expect(html()).toMatchSnapshot();
    expect(getByTestId("workflow-wizard")).toBeEmptyDOMElement();
  });
  it("renders a list of steps", () => {
    const routes = [
      {
        path: "/",
        name: "step-1"
      },
      {
        path: "step-2",
        name: "step-2"
      }
    ];

    const { getByText } = renderWithDependencies(WorkflowWizard, {
      routes,
      props: {
        steps: [
          {
            title: "Step 1",
            link: "step-1",
            disabled: false
          },
          {
            title: "Step 2",
            link: "step-2",
            disabled: false
          }
        ]
      }
    });

    const step1 = getByText("Step 1");
    const step2 = getByText("Step 2");

    expect(step1).toBeInTheDocument();
    expect(step1).toHaveAttribute("href", "#/");
    expect(step1).toHaveClass("workflow-wizzard__step--active");

    expect(step2).toBeInTheDocument();
    expect(step2).toHaveAttribute("href", "#step-2");
  });

  it("renders a disabled step", () => {
    const routes = [
      {
        path: "step-1",
        name: "step-1"
      }
    ];

    const { getByText } = renderWithDependencies(WorkflowWizard, {
      routes,
      props: {
        steps: [
          {
            title: "Step 1",
            link: "step-1",
            disabled: true
          }
        ]
      }
    });

    const step1 = getByText("Step 1");

    expect(step1).toBeInTheDocument();
    expect(step1).not.toHaveAttribute("href");
  });
});
