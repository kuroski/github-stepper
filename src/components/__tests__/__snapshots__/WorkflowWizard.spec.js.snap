// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowWizard renders a empty list of steps 1`] = `<div data-testid="workflow-wizard"></div>`;
